libs/barista-components/filter-field/src/filter-field-multi-select/filter-field-multi-select-trigger.ts

```diff
@@ -0,0 +1,181 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  DOWN_ARROW,
+  ENTER,
+  ESCAPE,
+  SPACE,
+  TAB,
+  UP_ARROW,
+} from '@angular/cdk/keycodes';
+import { Overlay, OverlayContainer, ViewportRuler } from '@angular/cdk/overlay';
+import { Platform } from '@angular/cdk/platform';
+import { DOCUMENT } from '@angular/common';
+import {
+  ChangeDetectorRef,
+  Directive,
+  ElementRef,
+  Inject,
+  Input,
+  NgZone,
+  OnDestroy,
+  Optional,
+} from '@angular/core';
+import {
+  DtOption,
+  DtViewportResizer,
+  _countGroupLabelsBeforeOption,
+  _getOptionScrollPosition,
+  _readKeyCode,
+} from '@dynatrace/barista-components/core';
+import { DtFilterFieldElementTrigger } from '../shared/filter-field-element-trigger';
+import { DtFilterFieldMultiSelect } from './filter-field-multi-select';
+
+@Directive({
+  selector: `input[dtFilterFieldMultiSelect]`,
+  exportAs: 'dtFilterFieldMultiSelectTrigger',
+})
+export class DtFilterFieldMultiSelectTrigger<T>
+  extends DtFilterFieldElementTrigger<DtFilterFieldMultiSelect<T>>
+  implements OnDestroy {
+  /** The filter-field multiSelect panel to be attached to this trigger. */
+  @Input('dtFilterFieldMultiSelect')
+  get element(): DtFilterFieldMultiSelect<T> {
+    return this._element;
+  }
+  set element(value: DtFilterFieldMultiSelect<T>) {
+    super.element = value;
+  }
+
+  /**
+   * Whether the autocomplete is disabled. When disabled, the element will
+   * act as a regular input and the user won't be able to open the panel.
+   */
+  @Input('dtFilterFieldMultiSelectDisabled')
+  get elementDisabled(): boolean {
+    return this._elementDisabled;
+  }
+  set elementDisabled(value: boolean) {
+    super.elementDisabled = value;
+  }
+
+  constructor(
+    protected _elementRef: ElementRef,
+    protected _overlay: Overlay,
+    protected _changeDetectorRef: ChangeDetectorRef,
+    protected _viewportRuler: ViewportRuler,
+    protected _platform: Platform,
+    protected _overlayContainer: OverlayContainer,
+    _zone: NgZone,
```

Comment:
Nitpick: 
Not quite necessary to prefix this with a `_` as this is not a class member